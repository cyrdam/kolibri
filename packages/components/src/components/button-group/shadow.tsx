import type { ButtonGroupProps } from '../../schema';
import type { JSX } from '@stencil/core';
import { Component, h, Host } from '@stencil/core';
import { KolButtonGroupWcTag } from '../../core/component-names';

@Component({
	tag: 'kol-button-group',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolButtonGroup implements ButtonGroupProps {
	public render(): JSX.Element {
		return (
			<Host class="kol-button-group">
				<KolButtonGroupWcTag>
					<slot />
				</KolButtonGroupWcTag>
			</Host>
		);
	}
}
