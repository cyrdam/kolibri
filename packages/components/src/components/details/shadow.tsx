import { Component, Element, h, type JSX, Method, Prop, State, Watch } from '@stencil/core';
import type { DetailsAPI, DetailsCallbacksPropType, DetailsStates, DisabledPropType, FocusableElement, HeadingLevel, LabelPropType } from '../../schema';
import { validateDetailsCallbacks, validateDisabled, validateLabel, validateOpen } from '../../schema';
import KolCollapsibleFc, { type CollapsibleProps } from '../../functional-components/Collapsible';
import { nonce } from '../../utils/dev.utils';
import { watchHeadingLevel } from '../heading/validation';
import { dispatchDomEvent, KolEvent } from '../../utils/events';

/**
 * @slot - Der Inhalt, der in der Detailbeschreibung angezeigt wird.
 */
@Component({
	tag: 'kol-details',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolDetails implements DetailsAPI, FocusableElement {
	@Element() private readonly host?: HTMLKolDetailsElement;

	private readonly nonce = nonce();
	private buttonWcRef?: HTMLKolButtonWcElement;

	private readonly catchRef = (ref?: HTMLKolButtonWcElement) => {
		this.buttonWcRef = ref;
	};

	/**
	 * @deprecated Use kolFocus instead.
	 */
	@Method()
	public async focus() {
		await this.kolFocus();
	}

	@Method()
	// eslint-disable-next-line @typescript-eslint/require-await
	public async kolFocus() {
		await this.buttonWcRef?.kolFocus();
	}

	private toggleTimeout?: ReturnType<typeof setTimeout>;

	private handleOnClick = (event: MouseEvent) => {
		this._open = !this._open;

		/**
		 * Der Timeout wird benötigt, damit das Event
		 * vom Button- auf das Accordion-Event wechselt.
		 * So ist es dem Anwendenden möglich das _open-
		 * Attribute abzufragen.
		 */

		clearTimeout(this.toggleTimeout);

		this.toggleTimeout = setTimeout(() => {
			if (this.host) {
				dispatchDomEvent(this.host, KolEvent.toggle, Boolean(this._open));
			}
			this.state._on?.onToggle?.(event, Boolean(this._open));
		}, 25);
	};

	public render(): JSX.Element {
		const { _open, _label, _disabled } = this.state;
		const _level = 1;

		const rootClass = 'kol-details';

		const props: CollapsibleProps = {
			id: this.nonce,
			label: _label,
			open: _open,
			disabled: _disabled,
			level: _level,
			onClick: this.handleOnClick,
			class: rootClass,
			HeadingProps: { class: `${rootClass}__heading` },
			HeadingButtonProps: {
				ref: this.catchRef,
				class: `${rootClass}__heading-button`,
				_icons: 'codicon codicon-chevron-right',
			},
			ContentProps: {
				class: `${rootClass}__content indented-text`,
				wrapperClass: `${rootClass}__wrapper`,
				animationClass: `${rootClass}__wrapper-animation`,
			},
		};

		return (
			<KolCollapsibleFc {...props}>
				<slot />
			</KolCollapsibleFc>
		);
	}

	/**
	 * Makes the element not focusable and ignore all events.
	 */
	@Prop() public _disabled?: boolean = false;

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: LabelPropType;

	/**
	 * Defines which H-level from 1-6 the heading has. 0 specifies no heading and is shown as bold text.
	 */
	@Prop() public _level?: HeadingLevel = 1;

	/**
	 * Defines the callback functions for details.
	 */
	@Prop() public _on?: DetailsCallbacksPropType<boolean>;

	/**
	 * If set (to true) opens/expands the element, closes if not set (or set to false).
	 * @TODO: Change type back to `OpenPropType` after Stencil#4663 has been resolved.
	 */
	@Prop({ mutable: true, reflect: true }) public _open?: boolean = false;

	@State() public state: DetailsStates = {
		_label: '', // ⚠ required
		_level: 1,
		_on: {},
	};

	@Watch('_disabled')
	public validateDisabled(value?: DisabledPropType): void {
		validateDisabled(this, value);
	}

	@Watch('_label')
	public validateLabel(value?: LabelPropType): void {
		validateLabel(this, value, {
			required: true,
		});
	}

	@Watch('_level')
	public validateLevel(value?: HeadingLevel): void {
		watchHeadingLevel(this, value);
	}

	@Watch('_on')
	public validateOn(on?: DetailsCallbacksPropType<boolean>) {
		validateDetailsCallbacks(this, on);
	}

	@Watch('_open')
	public validateOpen(value?: boolean): void {
		validateOpen(this, value);
	}

	public componentWillLoad(): void {
		this.validateDisabled(this._disabled);
		this.validateLabel(this._label);
		this.validateLevel(this._level);
		this.validateOn(this._on);
		this.validateOpen(this._open);
	}
}
