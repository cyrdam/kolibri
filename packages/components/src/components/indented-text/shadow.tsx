import type { IndentedTextProps } from '../../schema';
import type { JSX } from '@stencil/core';
import { Component, h, Host } from '@stencil/core';
import { KolIndentedTextWcTag } from '../../core/component-names';

/**
 * @deprecated Will be removed in v3. Replace with custom HTML and styling.
 * @slot - Der Text.
 */
@Component({
	tag: 'kol-indented-text',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolIndentedText implements IndentedTextProps {
	public render(): JSX.Element {
		return (
			<Host class="kol-indented-text">
				<KolIndentedTextWcTag>
					<slot />
				</KolIndentedTextWcTag>
			</Host>
		);
	}
}
