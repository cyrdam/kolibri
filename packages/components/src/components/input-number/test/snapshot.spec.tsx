import { executeTests } from 'stencil-awesome-test';

import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';

import { getInputNumberHtml } from './html.mock';

import type { SpecPage } from '@stencil/core/testing';
import type { InputNumberProps } from '../../../schema';
import { KolInputNumber } from '../shadow';

executeTests<InputNumberProps>(
	'InputNumber',
	async (props): Promise<SpecPage> => {
		const page = await newSpecPage({
			components: [KolInputNumber],
			template: () => <kol-input-number {...props} />,
		});
		return page;
	},
	{
		_label: ['Label'],
		_alert: [true, false],
		_icons: [[{ left: 'codicon codicon-home' }]],
		_readOnly: [true, false],
		_required: [true, false],
		_touched: [true, false],
		_max: [10],
		_min: [10],
		_step: [1, 2],
	},
	getInputNumberHtml,
	{
		execMode: 'default', // ready
		needTimers: true,
	},
);
