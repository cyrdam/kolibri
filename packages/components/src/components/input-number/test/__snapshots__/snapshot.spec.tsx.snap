// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _accessKey="V" 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          Label
          <span aria-hidden="true" class="access-key-hint">
            V
          </span>
        </span>
      </label>
      <div class="input">
        <input accesskey="V" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _alert=true 1`] = `
<kol-input-number _alert="">
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _disabled=true 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="disabled kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" disabled="" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _hint="Hint" 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <span class="hint" id="id-nonce-hint">
        Hint
      </span>
      <div class="input">
        <input aria-describedby="id-nonce-hint" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="icon-left icon-right input">
        <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label="" _tabindex="-1"></kol-icon>
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label="" _tabindex="-1"></kol-icon>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="icon-left input">
        <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label="" _tabindex="-1"></kol-icon>
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="icon-right input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label="" _tabindex="-1"></kol-icon>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideError=true 1`] = `
<kol-input-number _hide-error="">
  <template shadowrootmode="open">
    <div class="hidden-error kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _readOnly=true 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number read-only" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" readonly="" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _shortKey="S" 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          Label
          <span aria-hidden="true" class="access-key-hint">
            S
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _smartButton={"_icons":["codicon codicon-eye"],"_hideLabel":true,"_label":"einblenden"} 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <kol-button-wc _hidelabel="" _label="einblenden" _tabindex="-1" tabindex="0"></kol-button-wc>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _accessKey="V" 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          Label
          <span aria-hidden="true" class="access-key-hint">
            V
          </span>
        </span>
      </label>
      <div class="input">
        <input accesskey="V" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _alert=true 1`] = `
<kol-input-number _alert="">
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _disabled=true 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="disabled kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" disabled="" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _hint="Hint" 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <span class="hint" id="id-nonce-hint">
        Hint
      </span>
      <div class="input">
        <input aria-describedby="id-nonce-hint" autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _icons={"left":{"icon":"codicon codicon-arrow-left"},"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="icon-left icon-right input">
        <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label="" _tabindex="-1"></kol-icon>
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
        <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label="" _tabindex="-1"></kol-icon>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _icons={"left":{"icon":"codicon codicon-arrow-left"}} 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="icon-left input">
        <kol-icon _icons="codicon codicon-codicon codicon-arrow-left" _label="" _tabindex="-1"></kol-icon>
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _icons={"right":{"icon":"codicon codicon-arrow-right"}} 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="icon-right input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
        <kol-icon _icons="codicon codicon-codicon codicon-arrow-right" _label="" _tabindex="-1"></kol-icon>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} _hideError=true 1`] = `
<kol-input-number _hide-error="">
  <template shadowrootmode="open">
    <div class="hidden-error kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _msg={"_type":"error","_description":"Es ist ein Fehler aufgetreten"} 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _readOnly=true 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number read-only" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" readonly="" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _shortKey="S" 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          Label
          <span aria-hidden="true" class="access-key-hint">
            S
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _smartButton={"_icons":["codicon codicon-eye"],"_hideLabel":true,"_label":"einblenden"} 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
        <kol-button-wc _hidelabel="" _label="einblenden" _tabindex="-1" tabindex="0"></kol-button-wc>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] _touched=true 1`] = `
<kol-input-number _touched="">
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number touched" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _suggestions=[1,2,3] 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" list="id-nonce-list" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
        <datalist id="id-nonce-list">
          <option value="1"></option>
          <option value="2"></option>
          <option value="3"></option>
        </datalist>
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" _touched=true 1`] = `
<kol-input-number _touched="">
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number touched" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;

exports[`kol-input-number should render with _label="Label" _name="field" _placeholder="Hier steht ein Platzhaltertext" 1`] = `
<kol-input-number>
  <template shadowrootmode="open">
    <div class="kol-input kol-input-number msg-type-error number" role="presentation">
      <label class="input-label" htmlfor="id-nonce" id="id-nonce-label">
        <span class="input-label-span">
          <span>
            Label
          </span>
        </span>
      </label>
      <div class="input">
        <input autocapitalize="off" autocomplete="off" autocorrect="off" id="id-nonce" max="" min="" name="field" placeholder="Hier steht ein Platzhaltertext" type="number" value="">
      </div>
    </div>
  </template>
</kol-input-number>
`;
